src/rubrix/client/datasets.py

```diff
@@ -802,44 +807,74 @@ def prepare_for_training(self) -> "datasets.Dataset":
-            return datasets.Dataset.from_dict({})
-
-        class_tags = ["O"]
-        class_tags.extend(
-            [
-                f"{pre}-{label}"
-                for label in sorted(self.__all_labels__())
-                for pre in ["B", "I"]
-            ]
-        )
-        class_tags = datasets.ClassLabel(names=class_tags)
+            import datasets
 
-        def spans2iob(example):
-            r = TokenClassificationRecord(
-                text=example["text"],
-                tokens=example["tokens"],
-                annotation=self.__entities_to_tuple__(example["annotation"]),
+            has_annotations = False
+            for rec in self._records:
+                if rec.annotation is not None:
+                    has_annotations = True
+                    break
+
+            if not has_annotations:
+                return datasets.Dataset.from_dict({})
+
+            class_tags = ["O"]
+            class_tags.extend(
+                [
+                    f"{pre}-{label}"
+                    for label in sorted(self.__all_labels__())
+                    for pre in ["B", "I"]
+                ]
             )
-            return class_tags.str2int(r.spans2iob(r.annotation))
+            class_tags = datasets.ClassLabel(names=class_tags)
 
-        ds = (
-            self.to_datasets()
-            .filter(self.__only_annotations__)
-            .map(lambda example: {"ner_tags": spans2iob(example)})
-        )
-        new_features = ds.features.copy()
-        new_features["ner_tags"] = [class_tags]
+            def spans2iob(example):
+                r = TokenClassificationRecord(
+                    text=example["text"],
+                    tokens=example["tokens"],
+                    annotation=self.__entities_to_tuple__(example["annotation"]),
+                )
+                return class_tags.str2int(r.spans2iob(r.annotation))
+
+            ds = (
+                self.to_datasets()
+                .filter(self.__only_annotations__)
+                .map(lambda example: {"ner_tags": spans2iob(example)})
+            )
+            new_features = ds.features.copy()
+            new_features["ner_tags"] = [class_tags]
+
+            return ds.cast(new_features)
+
+        elif framework == "spacy":
+
+            import spacy
+            from spacy.tokens import DocBin
+
+            has_annotations = False
+            for rec in self._records:
+                if rec.annotation is not None:
+                    has_annotations = True
+                    break
+
+            nlp = spacy.blank("es")
+
+            db = DocBin()
+
+            # Creating the docbin object as in https://spacy.io/usage/training#training-data
+            for record in self._records:
+                doc = nlp(record.text)
+                ents = []
+                for anno in record.annotation:
+                    span = doc.char_span(anno[1], anno[2], label=anno[0])
+                    ents.append(span)
```

Comment:
Since we're building the spaCy doc without regarding the record tokenization (rubrix records contain info about text tokenization, and annotation spans are based on that tokenization), we should check if the resulting spaCy span is still valid (not`None`). If not, we should raise a warning flag, and discard it from the entities list.
```suggestion
                    span = doc.char_span(anno[1], anno[2], label=anno[0])
                    # There is a misalignment between record tokenization and spaCy tokenization
                    if span is None: 
                       print("Something wrong with record span and spaCy tokenization")
                    else:
                       ents.append(span)
```